From 73b029875248182f635085c1340b507a06c2a29f Mon Sep 17 00:00:00 2001
From: Chad Versace <chadversary@chromium.org>
Date: Fri, 28 Jun 2019 11:19:38 -0700
Subject: [PATCH] anv: Clamp apiVersion to 1.0.0 on Android <= Oreo

Android Pie accepts Vulkan 1.1. But the Nougat and Oreo CTS reject
Vulkan 1.1 in test
android.graphics.cts.VulkanFeaturesTest#testVulkanHardwareFeatures.

See <https://b.corp.google.com/issues/136215923>.
---
 src/intel/vulkan/anv_extensions_gen.py | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/src/intel/vulkan/anv_extensions_gen.py b/src/intel/vulkan/anv_extensions_gen.py
index a140c267452..233ead6823d 100644
--- a/src/intel/vulkan/anv_extensions_gen.py
+++ b/src/intel/vulkan/anv_extensions_gen.py
@@ -158,8 +158,17 @@ anv_physical_device_api_version(struct anv_physical_device *device)
 %for version in API_VERSIONS:
     if (!(${version.enable}))
         return version;
+
     version = ${version.version.c_vk_version()};
 
+#if defined(ANDROID_API_LEVEL) && ANDROID_API_LEVEL < 28
+    /* Limit the version 1.0 on Android earlier than Pie. The Nougat and Oreo
+     * CTS reject Vulkan 1.1 in test
+     * android.graphics.cts.VulkanFeaturesTest#testVulkanHardwareFeatures.
+     */
+    version = MIN2(version, VK_MAKE_VERSION(1, 0, 0));
+#endif
+
 %endfor
     return version;
 }
-- 
2.22.0.410.gd8fdbe21b5-goog

